### ISSUE TITLE
GMT_INSTALL_RELOCATABLE doesn't work as documented

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#847]
How to reproduce the issue:

1. `set (GMT_INSTALL_RELOCATABLE TRUE)` in cmake/ConfigUser.cmake
2. Compile and install GMT to `/opt/GMT-dev`, gmt works as expected.
3. Move `/opt/GMT-6.0.0` to `/opt/GMT-dev-new`, gmt doesn't print any copyright information anymore.

The output of `gmt -Vd` is:
```
gmt [DEBUG]: Obtained the ppid from parent: 75252
gmt [DEBUG]: Enter: gmtinit_new_GMT_ctrl
gmt [DEBUG]: GMT->session.SHAREDIR = /opt/GMT-dev/share
gmt [DEBUG]: GMT->session.HOMEDIR = /Users/seisman
gmt [DEBUG]: GMT->session.USERDIR = /Users/seisman/.gmt [created]
gmt [DEBUG]: GMT->session.CACHEDIR = /Users/seisman/.gmt/cache [created]
gmt [DEBUG]: GMT: 0. Will try to find subdir=postscriptlight stem = PSL_custom_fonts suffix=.txt
gmt [DEBUG]: GMT: 1. gmt_getsharepath trying current dir
gmt [DEBUG]: GMT: 2. gmt_getsharepath trying USERDIR /Users/seisman/.gmt
gmt [DEBUG]: Map distance calculation will be Cartesian
gmt [DEBUG]: Exit:  gmtinit_new_GMT_ctrl
gmt [DEBUG]: Enter: New_PSL_Ctrl
gmt [DEBUG]: Exit:  New_PSL_Ctrl
gmt [DEBUG]: Enter: gmt_manage_workflow
gmt [DEBUG]: Exit : gmt_manage_workflow
gmt [DEBUG]: Enter: PSL_beginsession
```
The reason is SHAREDIR isn't "relocatable". Manually set `GMT_SHAREDIR` to `/opt/GMT-dev-new/share` can fix the issue, but it's not ideal.

**Comment by PaulWessel** [comment]
Sorry, I do not know how we automate this. Can we get the executable path and figure it out from there?

**Comment by seisman** [comment]
This issue was fixed by #983
